### PR TITLE
Construct for form paths

### PR DESCRIPTION
**Pull request by Rahien**
This creates a construct query to fetch information necessary to render an instance of a form definition.

That construct query can then be used for:
- fetching all data for a form from a mu-auth aware triple store (SEAS)
- fetching only the form data from a form instance posted by the frontend
- deleting all form data from the backend and replacing it with the new form content (this is a little hardcore, we may want to be nicer and only replace the delta)

### CODE REVIEW

File: queries/getPathsForFields.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      ?field sh:path ?path.
+      # mid or last step of a complex path (blank node)
+      ?path rdf:rest+ ?step.
+      FILTER(?step != rdf:nil)
+      ?step rdf:first ?node.
+      ?previous rdf:rest ?step.
+      OPTIONAL {
+        ?node ?modifier ?predicate.
+      }
+    } ORDER BY ?field ?step
```
Comment (bfidlers) [inline queries/getPathsForFields.ts:187]: overbodig?

File: queries/getPathsForGenerators.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+  VALUES (?field) { ( <http://mu.semte.ch/vocabularies/core/uuid> ) }
+  `;
+
+  const paths = await Promise.all([
+    getSimplePaths(generatorWhere, formStore),
+    getPathHeads(generatorWhere, formStore),
+    getPathTails(generatorWhere, formStore),
+    getSimplePaths(uuidWhere, formStore),
+    getPathHeads(uuidWhere, formStore),
+    getPathTails(uuidWhere, formStore),
```
Comment (bfidlers) [inline queries/getPathsForGenerators.ts:189]: extra case necessary?

File: form-validator.ts [hunk 3]
```diff
@@ -1,20 +1,179 @@
...
+        predicate,
+        node: step || predicate, // step is undefined if there is no additional scope linked to the generator
+      };
+    } else {
+      previousToNext[previous] = {
+        predicate,
+        node: step,
+      };
+    }
+  });
```
Comment (bfidlers) [inline form-validator.ts:69]: Shared code with getPathsForFields, look what can be reused with functions.

**Review by bfidlers — APPROVED** [review]
Few small fixes, but otherwise looks good.

File: queries/getPathsForFields.ts [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+
+    SELECT
+    ?field ?previous ?step ?node ?modifier ?predicate
+    WHERE {
+      ?field a form:Field.
+      ?field sh:path ?path.
+
+      {
+        {
```
[the remark this answers is not in the document]
Comment (Rahien) [reply queries/getPathsForFields.ts:42]: there is indeed a case missing for modified simple paths, but i thought it through and:
- we currently don't use modified simple paths e.g. `[ sh:inversePath foaf:name ]`
- If we want to use such paths, they can always be written in their complex form `( [ sh:inversePath foaf:name] )`

so i decided not to implement this after all

**Pull request merged**